src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/ServiceBufferFileStreamReader.cs

```diff
@@ -202,12 +202,36 @@ public IList<DbCellValue> ReadRow(long fileOffset, long rowId, IEnumerable<DbCol
                 }
                 FileStreamReadResult result = readFunc(currentFileOffset, rowId, column);
                 currentFileOffset += result.TotalLength;
+                result.Value.ExecutionPlanFileExtension = GetDbCellValueExecutionPlanFileExtension(result.Value, column);
                 results.Add(result.Value);
             }
 
             return results;
         }
 
+        /// <summary>
+        /// Returns the file extension for XMl values that represent an execution plan.
+        /// </summary>
+        /// <param name="cellValue">The DBCellValue containing XML execution plan data.</param>
+        /// <param name="column">The DbColumnWrapper instance that corresponds with the DbCellValue</param>
+        /// <returns>The corresponding file extension for the execution plan or an empty string if not applicable.</returns>
+        private string GetDbCellValueExecutionPlanFileExtension(DbCellValue cellValue, DbColumnWrapper column)
+        {
+            if (!column.IsXml && !column.IsJson)
+            {
+                return string.Empty;
+            }
+
+            var planFileExtension = string.Empty;
+
+            if (cellValue.DisplayValue.Contains("ShowPlanXML")) // Should xmlns value be used instead? "http://schemas.microsoft.com/sqlserver/2004/07/showplan"
```

Comment:
Question: SQL execution plan XML files contain a root XML node named "ShowPlanXML". Is it okay to use this value or should the XML namespace value be used instead to determine the file extension the Plan Viewer in ADS should use to view the graphical representation of the data?

Reply:
@alanrenmsft, this is an enhancement we wanted to add to query execution plans in ADS. There's no issue associated with this PR. I can make one if needed.